### PR TITLE
[beef-tool] Create workflow for beef-tool executable

### PR DESCRIPTION
**Pull request by HWXLR8**
Closes https://github.com/HWXLR8/beef-board/issues/104

@ASleepyCat Please give the generated executable a try on your machine, I only have 1 windows system to validate against.

### CODE REVIEW

File: .github/workflows/beef-tool.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          pip install pyinstaller
+
+      - name: Package beef-tool
+        run: |
+          pyinstaller --onefile --noconsole beef-tool/beef-tool.py
+
+      - name: Upload executable
+        uses: actions/upload-artifact@v4
+        with:
+          name: beef-tool.exe
```
Comment (ASleepyCat) [inline .github/workflows/beef-tool.yml:28]: The artifact is a zip file, so it gets named `beef-tool.exe.zip`. I would remove the `.exe` extension.

**Review by ASleepyCat — CHANGES_REQUESTED** [review]
1. When I try to run the tool, I get a false positive from Windows AV, thinking it's a trojan (`Trojan:Win32/Wacatac.B!ml`). I think adding a notice would be a good idea.
2. The tool doesn't work without `avrdude/avrdude.exe` being present in the same directory as the tool. That should be added to the guide.
3. `avrdude` doesn't automatically exit out of DFU, so you should also mention to replug the board from the computer after flashing.

**Comment by ASleepyCat** [comment]
Should we also point to QMK Toolbox as an alternative?

**Comment by HWXLR8** [comment]
>     2. The tool doesn't work without `avrdude/avrdude.exe` being present in the same directory as the tool. That should be added to the guide.

Currently experimenting with ways to embed `avrdude` into the executable.

**Comment by HWXLR8** [comment]
@ASleepyCat Addressed all comments

**Review by ASleepyCat — COMMENTED** [review]
Just a few more things I've noticed:

1. Instead of creating a new workflow, I would create a new build job inside `ci.yml` so that we don't have two pipelines running on every push ([like in this example](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-jobs-in-a-workflow)).
2. What happens if the `libusb` drivers aren't installed? Are people still going to have to clone/download this repo to install them?

**Comment by HWXLR8** [comment]
> Just a few more things I've noticed:
> 
>     1. Instead of creating a new workflow, I would create a new build job inside `ci.yml` so that we don't have two pipelines running on every push ([like in this example](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-jobs-in-a-workflow)).
Agreed.
>     2. What happens if the `libusb` drivers aren't installed? Are people still going to have to clone/download this repo to install them?
Thoughts on including the drivers with the package generated by the workflow?

**Comment by ASleepyCat** [comment]
> Thoughts on including the drivers with the package generated by the workflow?

I think that's fine, if you include the `README` file as well for instructions.

**Comment by HWXLR8** [comment]
Done

**Review by ASleepyCat — APPROVED** [review]
lgtm

**Pull request merged**